fix: Pre-commit mypy hook fails on pre-existing 'Class cannot subclass' errors

## Description
The pre-commit mypy hook fails on staged files due to pre-existing `[misc]` errors from torch and pydantic stubs:

```
src/minivess/config/models.py:38:18: error: Class cannot subclass "BaseModel" (has type "Any")  [misc]
src/minivess/adapters/sam2_decoder.py:35:23: error: Class cannot subclass "Module" (has type "Any")  [misc]
```

These errors affect ALL files that subclass `nn.Module`, `BaseModel`, or `ModelAdapter` — i.e., virtually every adapter and config file in the repo. They are NOT new errors; they exist on `main` and have been present since the files were created.

## Root Cause
The `.pre-commit-config.yaml` mypy hook only has `additional_dependencies: [pydantic>=2.10]` but lacks torch stubs. Without proper stubs, `nn.Module` and `BaseModel` resolve to `Any`, triggering the `[misc]` error under strict mode.

## Proposed Fix
One of:
1. **Add `--disable-error-code=misc`** to the mypy hook args (simplest, least invasive)
2. **Add `torch>=2.5` to `additional_dependencies`** (proper fix, but heavier)
3. **Configure `[[tool.mypy.overrides]]`** for torch/pydantic modules with `follow_imports = skip`

Option 1 is recommended since `[misc]` is a catch-all error code that mainly fires on stub issues in this repo.

## Affected Files (pre-existing)
- `src/minivess/config/models.py` — 11 errors (all BaseModel subclass)
- `src/minivess/adapters/*.py` — 2+ errors per file (all nn.Module subclass)
- `tests/v2/unit/*.py` — errors in test helper classes

## Workaround
Currently using `SKIP=mypy` to bypass the hook for commits that don't introduce new mypy errors.

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -6,12 +6,17 @@ repos:
         args: [--fix]
       - id: ruff-format
 
-  - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v1.14.1
+  - repo: local
     hooks:
       - id: mypy
-        additional_dependencies: [pydantic>=2.10]
-        args: [--ignore-missing-imports, --follow-imports=silent]
+        name: mypy
+        entry: uv run mypy
+        language: system
+        types: [python]
+        # Run on staged Python files only (uses project venv with all deps).
+        # mirrors-mypy ran in an isolated venv without torch/monai/pandera,
+        # causing false "Class cannot subclass Any" [misc] errors (#225).
+        pass_filenames: true
 
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v5.0.0

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -265,6 +265,15 @@ warn_return_any = false
 warn_unused_ignores = false
 disable_error_code = ["type-arg"]
 
+# Validation: pandera DataFrameModel resolves to Any (incomplete stubs),
+# and @pa.check / @pa.dataframe_check decorators are untyped.
+# whylogs ResultSet also lacks stubs (get_columns → object).
+[[tool.mypy.overrides]]
+module = ["minivess.validation.*"]
+disallow_subclassing_any = false
+disallow_untyped_decorators = false
+warn_return_any = false
+
 # Archive: SAM2 reference code (archived, not part of active codebase)
 [[tool.mypy.overrides]]
 module = ["_archive.*"]

--- a/src/minivess/validation/profiling.py
+++ b/src/minivess/validation/profiling.py
@@ -31,7 +31,7 @@ class DatasetProfileView:
 
     def get_columns(self) -> list[str]:
         """Return the column names in the profile."""
-        return list(self._view.get_columns().keys())
+        return list(self._view.get_columns().keys())  # type: ignore[attr-defined]
 
     @property
     def raw(self) -> object:
@@ -82,8 +82,8 @@ def compare_profiles(
     drifted: list[str] = []
     summaries: dict[str, dict[str, float]] = {}
 
-    ref_cols = reference._view.get_columns()
-    cur_cols = current._view.get_columns()
+    ref_cols = reference._view.get_columns()  # type: ignore[attr-defined]
+    cur_cols = current._view.get_columns()  # type: ignore[attr-defined]
 
     for col_name in ref_cols:
         if col_name not in cur_cols:

